Drop-In/src/main/java/com/braintreepayments/api/PaymentMethodClient.java

```diff
@@ -94,46 +96,52 @@ void getPaymentMethodNonces(GetPaymentMethodNoncesCallback callback) {
     }
 
     void deletePaymentMethod(final Context context, final PaymentMethodNonce paymentMethodNonce, final DeletePaymentMethodNonceCallback callback) {
-        boolean usesClientToken = braintreeClient.getAuthorization() instanceof ClientToken;
+        braintreeClient.getAuthorization(new AuthorizationCallback() {
+            @Override
+            public void onAuthorizationResult(@Nullable Authorization authorization, @Nullable Exception error) {
+                boolean usesClientToken = authorization instanceof ClientToken;
+
+                if (!usesClientToken) {
+                    Exception clientTokenRequiredError =
```

Comment:
This technically also handles when auth is null and there's an error, but do we want to return the explicit error in that case instead?

Reply:
This actually calls back. The other one was guarding a UI update for vaulted payment methods.